### PR TITLE
feat: use new custom env vars

### PR DESCRIPTION
**Pull request by mistydemeo**
These two new environment variables are defined for compatibility with cargo-dist, which uses the same variables in its installers: https://github.com/axodotdev/cargo-dist/pull/1503

### CODE REVIEW

File: axoupdater/src/errors.rs [hunk 1]
```diff
@@ -166,4 +170,26 @@ pub enum AxoupdateError {
     )]
     #[diagnostic(help("This probably isn't your fault; please open an issue at https://github.com/axodotdev/axoupdater!"))]
     CleanupFailed {},
+
+    /// User passed conflicting GitHub API environment variables
+    #[error("Both {ghe_env_var} and {github_env_var} have been set in the environment")]
```
Comment (ashleygwilliams) [inline axoupdater/src/errors.rs:175]: do we want to add the installer_download_url to this?

File: axoupdater/src/errors.rs [hunk 2]
```diff
@@ -166,4 +170,26 @@ pub enum AxoupdateError {
...
+    MultipleGitHubAPIs {
+        /// The GitHub Enterprise env var
+        ghe_env_var: String,
+        /// The GitHub env var
+        github_env_var: String,
+    },
+
+    /// Couldn't parse the text domain (could be an IP, etc.)
+    #[error("Unable to parse the domain from the passed url: {url}")]
+    #[diagnostic(help("The {env_var} variable only takes domains. If you're using an IP, we recommend the GitHub Enterprise-style variable: {ghe_env_var}"))]
```
Comment (ashleygwilliams) [inline axoupdater/src/errors.rs:186]: super nice

File: axoupdater/src/release/github.rs [hunk 3]
```diff
@@ -1,17 +1,44 @@
...
+    if env::var(&ghe_env_var).is_ok() && env::var(&github_env_var).is_ok() {
+        return Err(AxoupdateError::MultipleGitHubAPIs {
+            ghe_env_var,
+            github_env_var,
+        });
+    }
 
-fn github_api() -> String {
-    env::var("INSTALLER_BASE_URL").unwrap_or_else(|_| "https://api.github.com".to_string())
+    if let Ok(value) = env::var(&ghe_env_var) {
```
Comment (ashleygwilliams) [inline axoupdater/src/release/github.rs:26]: spiritually i feel like this wants to be a match not an if since we don't allow the 2 conditions

**Review by ashleygwilliams — APPROVED** [review]
lgtm!

File: axoupdater/src/release/github.rs (hunk 3 above)
Comment (fasterthanlime) [reply axoupdater/src/release/github.rs:26]: I don't see how to write this code as a match! I think it's fine as-is

File: axoupdater/src/lib.rs [hunk 4]
```diff
@@ -621,6 +621,11 @@ fn get_app_name() -> Option<String> {
     }
 }
 
+/// Returns an environment variable-compatible version of the app name.
+pub fn app_name_to_env_var(app_name: &str) -> String {
```
Comment (fasterthanlime) [inline axoupdater/src/lib.rs:625]: This helper is the morally correct thing to do, thanks for your service

**Review by fasterthanlime — APPROVED** [review]

File: axoupdater/src/errors.rs (hunk 1 above)
Comment (mistydemeo) [reply axoupdater/src/errors.rs:175]: I don't think so - we don't use it in the updater.

**Pull request merged**